OmsAgent/omsagent.py

```diff
@@ -1024,14 +1025,21 @@ def exit_if_gpg_unavailable(operation):
     If not, attempt to install
     If install fails, throw error to return UnsupportedGpg error code
     """
-    check_exit_code, _ = run_get_output('which gpg', True, False)
-    if check_exit_code is not 0:
-        hutil_log_info('GPG not found, attempting to install')
-        exit_code, output = run_get_output(InstallExtraPackageCommand.format('gpg'))
-        if exit_code is not 0:
-            log_and_exit(operation, UnsupportedGpg, 'GPG could not be installed: {0}'.format(output))
-        else:
-            hutil_log_info('GPG successfully installed')
+    # Check if VM is Debian 10 (temporary fix to unblock Deb 10)
+    vm_supp, vm_dist, vm_ver = is_vm_supported_for_extension()
+    if (vm_supp and (vm_dist.lower().startswith('debian')) and (vm_ver.startswith('10')):
```

Comment:
you can only check for debian , no need to check the version. We already know other images come with gpg installed and in case 11 needs install , you would not need to change this code. This gate is only needed to scope it to debian to remove the remaining piped commands from install cmdline.